common/predictive-text/unit_tests/headless/worker-custom-punctuation.js

```diff
@@ -8,16 +8,7 @@ var ModelCompositor = require('../../build/intermediate').ModelCompositor;
 
 describe('Custom Punctuation', function () {
   it('appears in the keep suggestion', function () {
-    let dummySuggestions = [
-      {
-        transform: {
-          insert: 'Hrllo',
-          deleteLeft: 0,
-        },
-        tag: 'keep',
-        displayAs: 'Hrllo',
-      },
```

Comment:
The 'keep' suggestion is automatically generated by the compositor. If the model returns it instead, the compositor assumes it to be an actual prediction.

This only resulted in a breakage now due to the new changes, though the test was ineffective for a while before this.

Reply:
The `tag: 'keep'` line is supposed to be added by the `ModelCompositor`, while the original line was returning it from the `DummyModel`'s suggestions.  Due to how its `Suggestion`s were parsed, the hardcoded `keep` was able to "pass through" the `ModelCompositor` unfazed until now.

This block was partly responsible:  

https://github.com/keymanapp/keyman/blob/1937ac80b42800da65502ec264b432ee6c28faa4/common/predictive-text/worker/model-compositor.ts#L219-L227

Before, the code would compare against an output suggestion's raw text and preserve the original `Suggestion`, mutating it into a completed "keep" `Suggestion` on the spot.  Importantly, a version of "keep" _without quote-like punctuation_.  The quote-punctuation was applied later:

https://github.com/keymanapp/keyman/blob/1937ac80b42800da65502ec264b432ee6c28faa4/common/predictive-text/worker/model-compositor.ts#L253-L265

Note how this block doesn't care what the source of the 'keep' `Suggestion` is - it will apply quotes to it _regardless_ of whether or not it has a corresponding prediction from the model.

Compare against this against the 'changed' block:

https://github.com/keymanapp/keyman/blob/cbf56266d522e07b7fd64079ee924fa35ae69a47/common/predictive-text/worker/model-compositor.ts#L212-L225

When combined with the other changes from this PR, the abstracted `toAnnotatedKeepSuggestion` handles "keep" punctuation processing and adds quotes _only_ if the "keep" is not available as a model prediction.

End result:

```
 1) Custom Punctuation
       appears in the keep suggestion:

      AssertionError: expected 'Hrllo' to equal '«Hrllo»'
      + expected - actual

      -Hrllo
      +«Hrllo»
      
      at Context.<anonymous> (unit_tests/headless/worker-custom-punctuation.js:60:12)
```

We don't want to show quotes when the 'keep' `Suggestion` matches a model prediction, and fixing that bug caused the test to fail.